factcast-core/src/main/java/org/factcast/core/util/FactCastJson.java

```diff
@@ -158,10 +177,21 @@ public static JsonNode toJsonNode(Map<String, Object> jsonAsMap) {
     @SneakyThrows
     public static <A> A readValueFromBytes(Class<A> type, byte[] bytes) {
         return objectMapper.readerFor(type).readValue(bytes);
-
     }
 
     public static ObjectMapper mapper() {
         return objectMapper;
     }
+
+    @SneakyThrows
+    public static long calculateHash(Class<?> projectionClass) {
+
+        JsonSchema jsonSchema = schemaGen.generateSchema(projectionClass);
+
+        String schema = writer
+                .writeValueAsString(jsonSchema);
+
+        return schemaModifier.apply(schema)
+                .hashCode();
```

Comment:
is the hashcode even guaranteed to be consistent? i'd expect taking the md5 of the string

Reply:
changed